### ISSUE TITLE
Creating a Tag Manager container raises a PHP 8.1 deprecation notice when the container name isn't a string

### ISSUE THREAD

**Issue by timnolte** [google/site-kit-wp#7809]
## Bug Description

https://github.com/google/site-kit-wp/blob/6b5733abdc43beb0e319f93aebf80e255d9ac435/includes/Modules/Tag_Manager.php#L253-L257

`sanitize_container_name()` receives the container name from `get_bloginfo( 'name' ) ?: URL::parse( $this->context->get_reference_site_url(), PHP_URL_HOST )`. `URL::parse()` returns `null` for a URL with no host. On a site with an empty **Site Title**, the name reaches `trim()` as `null`, and PHP 8.1 throws a deprecation notice:

```
Deprecated: trim(): Passing null to parameter #1 ($string) of type string is deprecated
```

`trim()` returns a string, so the `ltrim()` call below it receives an empty string and throws nothing.

`sanitize_container_name()` should check that the value is a string before it calls `trim()`.

## Steps to reproduce

1. On PHP 8.1, set `WP_DEBUG` and `WP_DEBUG_LOG` to `true` in `wp-config.php`.
2. Clear the **Site Title** field in **Settings > General** and save.
3. Add a `googlesitekit_site_url` filter that returns a value with no host, such as `example.com`.
4. Empty `wp-content/debug.log`.
5. Connect Tag Manager and create a new web container.
6. Read `wp-content/debug.log`.

Expected: Site Kit creates the container, and no deprecation notice names the Tag Manager module.

## Additional Context

- PHP Version: 8.1.29
- Plugin Version: 1.185.0

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* On PHP 8.1 or later, Site Kit raises no PHP deprecation notice when a Tag Manager container name isn't a string.
* Following the **steps to reproduce** creates the Tag Manager container, and the PHP error log holds no deprecation notice from Site Kit.
* A Tag Manager container name that's already a string comes out exactly as it does today. Site Kit still cuts surrounding whitespace, removes leading underscores and accents, decodes HTML entities, collapses repeated spaces, and keeps only the characters `a-zA-Z0-9_., -`.

## Implementation Brief

* [ ] 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by timnolte** [comment]
I'll get a PR opened for a fix for this. I've tested it and a simple validation before the call is all that is needed.

**Comment by kuasha420** [comment]
Thank you for raising the issue @timnolte ! Can you provide a step to reproduce this issue? Cheers!

**Comment by timnolte** [comment]
@adamdunnage this isn't in need of some additional support forum request. This is a code issue that needs to be fixed, which I already provided a PR to fix this. The output that @mrwweb provided are the details to traceback this depreciation notice. This is a PHP 8.1 compatibility issue not an end user issue.

**Comment by timnolte** [comment]
@mrwweb just noticed that your stack trace does appear to call out the `ltrim()` in WordPress Core. Have you taken a look at the code called out in each line of the stack trace? If not I can take a look to determine if I can update my PR to include fixing this as well. I'm suspecting that this may be partly do an incomplete plugin setup, and there isn't adequate validation happening.

**Comment by mrwweb** [comment]
@timnolte Haven't done anything else. I saw the error, found this issue, and saw the request for a stacktrace and dumped it here. Hope it's useful!

**Comment by kevinlisota** [comment]
This issue still exists in Site Kit. I get the deprecation warning in both PHP 8.1 and 8.2 on every page load with stack trace results like originally reported.

**Comment by bethanylang** [comment]
@kevinlisota Yes, this issue remains open at the moment, but it's on our list to fix in a future release. Feel free to follow along here for updates. Thanks!

**Comment by zutigrm** [comment]
Wasn't able to reproduce it, tested on InstaWP using PHP `8.1` and `8.2`, with and without Tag manager being active. Couldn;t see a scenario where `$name` would be empty 🤔 

Unassigning myself so someone else can try as well

**Comment by shervElmi** [comment]
Updated the title and description of the issue, and added acceptance criteria.

The deprecation comes from `trim()` in [`Tag_Manager.php`](https://github.com/google/site-kit-wp/blob/6b5733abdc43beb0e319f93aebf80e255d9ac435/includes/Modules/Tag_Manager.php#L255), not `ltrim()`. `trim()` returns a string, so the `ltrim()` call below it never receives `null`.
